memory/cloudsql/chat_message_history_container_test.go

```diff
@@ -0,0 +1,153 @@
+package cloudsql
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	tcpostgres "github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/util/cloudsqlutil"
+)
+
+type chatMsg struct{}
+
+func (chatMsg) GetType() llms.ChatMessageType {
+	return llms.ChatMessageTypeHuman
+}
+
+func (chatMsg) GetContent() string {
+	return "test content"
+}
+
+func preCheckEnvSetting(ctx context.Context, t *testing.T) string {
+	t.Helper()
+
+	if openaiKey := os.Getenv("OPENAI_API_KEY"); openaiKey == "" {
+		t.Skip("OPENAI_API_KEY not set")
+	}
+
+	pgvectorURL := os.Getenv("PGVECTOR_CONNECTION_STRING")
+	if pgvectorURL == "" {
+		pgVectorContainer, err := tcpostgres.RunContainer(
+			ctx,
+			testcontainers.WithImage("docker.io/pgvector/pgvector:pg16"),
+			tcpostgres.WithDatabase("db_test"),
+			tcpostgres.WithUsername("user"),
+			tcpostgres.WithPassword("passw0rd!"),
+			testcontainers.WithWaitStrategy(
+				wait.ForLog("database system is ready to accept connections").
+					WithOccurrence(2).
+					WithStartupTimeout(30*time.Second)),
+		)
+		if err != nil && strings.Contains(err.Error(), "Cannot connect to the Docker daemon") {
+			t.Skip("Docker not available")
+		}
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, pgVectorContainer.Terminate(ctx))
+		})
+
+		str, err := pgVectorContainer.ConnectionString(ctx, "sslmode=disable")
+		require.NoError(t, err)
+
+		pgvectorURL = str
```

Comment:
Is there a reason we are using pgvector + test_container here rather than connect to Cloud SQL?